---
layout: default
title: "Projects2018.txt"
tags: participants apply
---

# hackseq18 project list

{:toc}

1. Project 1: REUSE
2. ...

**Skill Proficiencies**<br/><br/>
 	\*       : "I can work with this langauge, if I have access to Stack Overflow ..."<br/><br/>
 	\* \*    : "I know this language and use it regularily."<br/><br/>
 	\* \* \* : "I am one with the language."<br/><br/>

----------------------------------------

## [Project 1: REUSE](https://github.com/chorltsd/REUSE)
**Team Lead:** [Sam Chorlton](samchorlton.com) (sam [AT] samchorlton.com)

**Abstract:**

Filtering unwanted sequences from nucleic acid sequencing data is an important step in many analyses. It has been used to remove technical artefacts (eg. PhiX), discover known and novel pathogens, isolate nucleic acid types (eg. rRNA), and remove noise in metagenomic studies. This step significantly improves the speed and quality of subsequent analyses. Commonly used approaches include mapping of reads to a reference set, or filtration via bloom trees or k-mer hashes. K-mers have been shown to accurately differentiate reads between species in a fraction of the time as traditional read mapping approaches; however, currently implemented approaches such as BBDuk, Kontaminant and Cookiecutter are limited by memory usage, parallelization and other practical features. Here, we develop REUSE, a program to Rapidly Eliminate Useless SEquences. REUSE implements a minimal perfect hash function to generate a reference index with limited RAM and time. Searching the index is performed using the complete k-mer set from each read, and reeads can be discarded or retained, depending on user preference, if they contain a pre-specified number of k-mers found in the index. In comparisons against other tools on simulated and real data, REUSE is consistently faster and uses less RAM. REUSE demonstrates similar accuracy to traditional read mappers, and produces identical results to other k-mer based tools. REUSE is publicly available at https://github.com/chorltsd/REUSE.

----------------------------------------

## [Project 2: Genome Assembler Components](https://github.com/hackseq/modular-assembly-hs18)
**Team Leads:** [Shaun Jackman](http://sjackman.ca/), Rayan Chikhi and Paul Medvedev

*Our hackseq project aims to create a better assembly tool by mixing and matching components of various assembly tools. We will pit these Frankenstein assembly pipelines against each other in a fantastic assembler bake off!*
 
**Abstract:**

Genome sequencing yields many short reads of DNA from a genome. Genome assembly attempts to reconstruct the original genome from these reads. Most genome assembler software tools are pipelines of many stages. It's not however typical to mix and match stages from different tools. Our Hackseq project aims to create a better assembly tool by mixing and matching components of various assembly tools.

We will work together to create modular assembler components for AssemblerFlow, a tool which builds pipelines of tools for genome assembly using Nextflow. Each participant will create and run a genome assembly pipeline using AssemblerFlow. We will assess the quality of each assembly, using Quast. Finally, we'll create a leader board of awesomeness to compare the assembly results!

----------------------------------------

## [Project 3: TaraCyc](https://github.com/hackseq/tara-cyc-hs18)
**Team Lead:** Arjun Baghela (ECOSCOPE)

*This project will aim to construct a website to represent thousands of ocean virus genomes. The goal of the website is to allow world citizens to play, explore and interact with the genome dataset to gain understanding into the microscopic life living in Earth’s oceans.*

**Abstract:**

TARA Oceans represents one of the world’s greatest coordinated research and sampling efforts and has collected years of standardized and detailed information about Earth’s largest cohesive eco-system providing insights crucial for the preservation of mankind our planet. Of particular focus is microbial community metabolism, which drives biogeochemical cycling in the ocean through networks of metabolite exchange and gene transfer. Viruses can reprogram these networks by co-opting host metabolism during infection using auxiliary metabolic genes (AMGs) encoded in viral genomes. Metabolic reprogramming has the potential to alter metabolic flux at the individual, population and community levels with resulting feedback on biogeochemical cycling at ecosystem scales. While previous gene-centric studies have identified viral AMGs associated with carbon and energy metabolism in the sunlit and dark ocean, the extent to which reprogramming targets the metabolic repertoire of marine microbial communities remains unconstrained. Here we use a pathway-centric approach to identify known and novel metabolic hotspots associated with global AMG distribution patterns and bound viral reprogramming potential in the ocean.

We used MetaPathways v2.5 in combination with Pathway Tools and the MetaCyc hierarchy of metabolic pathways to construct an environmental pathway genome database (ePGDBs) from hundreds of thousand viral contigs assembled from the Tara Oceans transect . This global ocean virome (GOV) was assembled from a subset of 90 viral fractions from a total of 288 size-fractioned samples spanning the entire transect. For Hackseq, we wish to construct a website called TaraCyc, which would represent a compendium of these ePGDBs and provide a comprehensive collection of MetaPathways output and extracted Environmental Pathway/Genome Databases (ePGDBs). Further, the TaraCyc website would allow world citizens to play, explore and interact with the dataset and gain understanding into the microscopic life living in Earth’s ocean.

----------------------------------------


## [Project 4: beRi: Virtual Environments for R](https://github.com/datasnakes/beRi)
**Team Lead:** Rob Gilmore (robgilmore127 [AT] gmail.com)

*Are you a bioinformatician frustrated by R's inability to resolve common dependency issues? Help develop beRi, a package management, reproducible workflow, and installation toolkit for the R programming language.*

**Abstract:**

The R programming language is a popular choice for statisticians, bioinformaticians, and researchers performing data analysis work.  However, it lacks management tools that ensure reproducibility and integration with command line interfaces (CLI).
  
*Problem:* R lacks proper management tools such as a project manager, a virtual environment manager, or a package manager.  Many available tools are R packages, making it laborious to manage R from the command line.  While Rstudio has developed packrat, a project management tool, it has performance issues including lengthy build times. packrat also works poorly with Bioconductor, one of the most widely used R package repositories outside of CRAN. Building an R CLI is also a daunting task even for advanced R users.

*Solution:* We propose the beRi suite of tools for managing R.  beRi is a set of Python packages which is composed of renv - an R virtual environment manager, rinse - an R installation manager and utility tool for installing packages, and rut - a manager for native R configuration files and local CRAN-like repositories.  Since these packages will be developed as standalone or fully independent CLI’s, beRi will integrate these three packages and complement their functionalities in a separate higher function interface.

----------------------------------------

## [Project 5: Alignment-free Pathogen Genomics](https://github.com/SweetiePi/bioncd-hackseq)
**Team Lead:** Alex Sweeten (alex.sweeten [AT] gmail.com)

*Interested in developing a novel bioinformatics pipeline? Given the unprecedented size of genomic information, new methods are required to organize and manage this data. The goal of this project will be to develop and test an alignment-free method, and apply it towards datasets of pathogenic organisms.*
 
**Abstract:**

Determining similarity between bacterial isolates is an important requirement for epidemiological analysis. Alignment-based genomic methods are commonly used to tackle this problem. However, in cases of low sequence homology, horizontal gene transfer, or lack of a priori information, as is common when dealing with pathogenic bacteria, alignment-based methods pose significant problems. The normalized compression distance (NCD) is a parameter and alignment-free distance metric, which has shown recent success in genomics, specifically in classifying viral sequences.
Here, I propose to develop a pipeline for allowing users to apply NCD to their genomic data, as well as for users to visualize their results in a presentable and easy to read format. This application will be built using a Python framework and developed into a Conda package and/or Galaxy workflow (depending on time required & participant experience). Furthermore, I propose to test/benchmark our pipeline using datasets of pathogenic bacteria and viral sequences. This project will be a novel community-based application, potentially facilitating further research into alignment-free similarity methods.

----------------------------------------

## [Project 6: Simulating Transcriptome Structural Variants](https://github.com/creisle/rna_sv_simulator)
**Team Leads:** Cara Reisle (caralynreisle [AT] gmail.com) & Morgan Bye (morgan [AT] morganbye.com)

*While there are tools to simulate structural variants in genomic data, currently no such applications exist for transcriptomes. Simulated data is required to properly evaluate structural variant callers.*

**Abstract:**

The goal of this project is to produce an application that is able to simulate structural variants in transcriptome data. The application will be written in python. It will be a pipeline consisting of several steps:
- Specification of the structural variants to be simulated
- Generation of the resulting transcriptome sequence
- Simulating paired-end reads covering the structural variant breakpoints
- Alignment of the simulated reads to the reference genome to produce a BAM file

If you’re new to structural variants or working with sequencing data, it would be a good idea to do a bit of background reading. I’ve listed a some potentially relevant literature below.
deFuse (PMID 21625565), Chimerascan (PMID 21840877), STAR (PMID 26334920), BAMSurgeon (PMID 25984700), Flux Simulator (PMID 22962361), MAVIS (PMID 30016509), SQUID (PMID 29650026)

----------------------------------------

## [Project 7: Blockchain for Infectious Disease](https://github.com/vghorakavi/InfectiousBlockchain)
**Team Lead:** Veena Ghorakavi (vghorakavi [AT] gmail.com)

*Infectious diseases are widespread and are difficult to track. Blockchain has the capabilities to track infectious diseases without revealing information about the people impacted and with location data.*
 
**Abstract:**

Understanding how to track and prevent the spread of infectious diseases is a task that needs to both protect the identity of the individual impacted and understand the prognosis of the spread. As a result, combining the tracking of infectious diseases with blockchain technology is an important albeit unexplored ideal. The idea is to allow people to mark themselves as infected and for data scientists and medical professionals to use this information.

----------------------------------------

## [Project 8: Anatomy of Morbidity](https://github.com/NoushinN/anatomy-of-morbidity)

**Team Lead:** [Noushin Nabavi](https://twitter.com/nabavinoushin) (nabavinoushin [AT] gmail.com)

*We are going to work together in order to analyze an open Government dataset released by Canadian Vital Statistics. We will make hypotheses about the causes of death across Canada over time and test our hypotheses through generating informative plots. This is a great opportunity for those who are new to computational biology to learn some new skills and meet other scientists in the field.*
 
**Abstract:**

With the rise of aging population and chronic diseases, Canada continues to suffer from the high costs of disease management [1-3]. Additionally, studies on prevalence and patterns of causes of morbidity or co-morbidity and associated determinants in Canada remain scarce and outdated [4]. Canadian Vital Statistics has recently released a comprehensive and well-annotated dataset on causes of death and mortality of all Canadians from 2012 until 2016. We hypothesize that understanding the major causes of mortality (i.e. various diseases) among males and females in different age categories across all provinces can inform health care and governmental policies. In this project, we will formulate hypotheses about what we most die of as a nation and test them by exploring these datasets. The overarching aim is to generate a final report on the most common causes of death across Canada that can at its best inform health care professionals and government agencies in developing evidence-base policies.
